packages/block-editor/src/hooks/use-zoom-out.js

```diff
@@ -21,23 +21,50 @@ export function useZoomOut( zoomOut = true ) {
 	);
 	const { isZoomOut } = unlock( useSelect( blockEditorStore ) );
 
-	useEffect( () => {
-		const isZoomOutOnMount = isZoomOut();
+	const toggleZoomOnUnmount = useRef( null );
+
+	// Let this hook know if the zoom state was changed manually.
+	const manualIsZoomOutCheck = isZoomOut();
 
+	useEffect( () => {
 		return () => {
-			if ( isZoomOutOnMount ) {
-				setZoomLevel( 'auto-scaled' );
-			} else {
+			if ( ! toggleZoomOnUnmount.current ) {
+				return;
+			}
+
+			if ( isZoomOut() ) {
 				resetZoomLevel();
+			} else {
+				setZoomLevel( 'auto-scaled' );
 			}
 		};
 	}, [] );
 
+	// This hook should only run when zoomOut changes, so we check for isZoomOut() within the
+	// hook rather than passing in
+	useEffect( () => {
+		const isZoomedOut = isZoomOut();
+
+		// Requested zoom and current zoom states are different, so toggle the state.
+		if ( zoomOut !== isZoomedOut ) {
```

Comment:
This check is so confusing unless you scroll up to see the function signature 😆

Reply:
Agreed. Not sure what should be renamed. I think the passed `zoomOut` would be best to rename. Maybe change to `setZoomState`? Or `zoomLevel`?